[aslref] Update build instructions (#1815)
herd/herdtools7: asllib/README.md

```bash
$ brew install opam
```

### Installing the release version

If you want to use the last released version in opam, simply use:
```bash
$ opam install aslref
```

To uninstall, simply run
```bash
$ opam remove aslref
```

### Installing from sources

1. Clone herdtools7:
    ```bash
    $ git clone https://github.com/herd/herdtools7.git
    ```
2. Ensure the opam environment is up-to-date. For example for bash:
   ```bash
   $ eval $(opam env)
   ```
3. Install dependencies:
   ```bash
   $ cd herdtools7
   $ dune build asllib/aslref.opam
   $ opam install ./asllib --deps-only
   ```
4. Build and install into a location `$PREFIX`:
   ```bash
   $ make build-aslref # optional
   $ make install-aslref PREFIX=$PREFIX
   ```

It's done!

To uninstall, you can remove the binaries from `$PREFIX`. You can also run:
```bash
$ make uninstall-aslref PREFIX=$PREFIX
```

### Checking

If `$PREFIX/bin` is in your `$PATH`, the following command should return a similar output:
```bash
$ aslref --version
aslref version 7.56+03 rev 7aa9d1f3cee2598ec64f14372f210e008ac5510f
```

## Usage

### Basics

If `my-test.asl` contains a valid ASL specification returning 0, the tool `aslref`

> does not print anything and exits with code 0.

Reviewer: Why is it important to say that the tool does not print anything?
Perhaps say "...the tool `aslref` interprets `my-test.asl` and exists with code 0."

Author: done.

```bash
$ aslref my-test.asl
```

> ### Version and type-checking flags

Reviewer: "Version" here refers to the ASL version, not the tool's version.
How about "ASL version and type-checking options"?

Author: I've removed this section following a comment by Hrutvik.

# Getting started with ASLRef

## Disclaimer

This material covers ASL1, the latest version of Architecture Specification
Language (ASL). ASL appears in the Arm Architecture Reference Manual:

> [developer.arm.com/documentation/ddi0487/latest](https://developer.arm.com/documentation/ddi0487/latest).

Reviewer 2: Perhaps:
```suggestion
Language (ASL). ASL appears in the [Arm Architecture Reference Manual](https://developer.arm.com/documentation/ddi0487/latest).
```
?

In addition to making contributions to this repository, Arm officially
maintains and regularly releases the ASL Reference, the formal language
definition for ASL1, available at

> [developer.arm.com/Architectures/Architecture Specification Language](https://developer.arm.com/Architectures/Architecture%20Specification%20Language).

Reviewer 2: ```suggestion
definition for ASL1, available on
[Arm Developer](https://developer.arm.com/Architectures/Architecture%20Specification%20Language).
```

We welcome feedback, questions, and feature requests — please contact us by
writing to [atg-formal@arm.com](mailto:atg-formal@arm.com) or raise issues and
pull requests to the herdtools7 GitHub repository.

## Installation

### Pre-requisites

> The following steps have been tested on Unix.

Reviewer 2: Should we remove this line?

> 1. Install ocaml and opam (ocaml package manager), see

Reviewer 2: ```suggestion
Install ocaml and opam (ocaml package manager), see
```

   ```bash
   $ brew install opam
   ```

### Installing the release version

If you want to use the last released version in opam, simply use:
```bash
$ opam install aslref
```

To uninstall, simply run
```bash
$ opam remove aslref
```

### Installing from sources

1. Clone herdtools7:
    ```bash
    $ git clone https://github.com/herd/herdtools7.git
    ```
2. Ensure the opam environment is up-to-date. For example for bash:
   ```bash
   $ eval $(opam env)
   ```
3. Install dependencies:
   ```bash
   $ cd herdtools7
   $ dune build asllib/aslref.opam
   $ opam install ./asllib --deps-only
   ```
4. Build and install into a location `$PREFIX`:
   ```bash
   $ make build-aslref # optional
   $ make install-aslref PREFIX=$PREFIX
   ```

It's done!

To uninstall, you can remove the binaries from `$PREFIX`. You can also run:
```bash
$ make uninstall-aslref PREFIX=$PREFIX
```

### Checking

If `$PREFIX/bin` is in your `$PATH`, the following command should return a similar output:
```bash
$ aslref --version
aslref version 7.56+03 rev 7aa9d1f3cee2598ec64f14372f210e008ac5510f
```

## Usage

### Basics

If `my-test.asl` contains a valid ASL specification returning 0, the tool `aslref`

> does not print anything and exit with code 0.

Reviewer: `does not print anything and exits with code 0.`

```bash
$ aslref my-test.asl
```

### Version and type-checking flags

For a complete reference of arguments, see `aslref --help`.

> #### ASL Version

Reviewer 2: Suggested to delete everything from here on in this PR, and then a separate PR can re-add if necessary. Perhaps contributing instructions should be in a `CONTRIBUTING.md` as well

Author: I've left the documentation for the command line flags, because it still applies. I've removed starting at examples because it was not up to date.
Does that satisfy your comment?

Reviewer 2: I would be in favour of removing all, and leaving the `--help` command to provide the necessary documentation (also ensuring that we don't forget to update documentation if we update command line options) - but happy to be convinced otherwise

Author: ok, done.
